s2spy/_resample.py

```diff
@@ -33,7 +33,7 @@ def mark_target_period(
     if isinstance(input_data, PandasData):
         input_data["target"] = np.zeros(input_data.index.size, dtype=bool)
         input_data["target"] = input_data["target"].where(
-            input_data["i_interval"] >= calendar.n_targets, other=True
+            input_data["i_interval"] < 0, other=True
```

Comment:
Didn't we agree that target is >0?  I don't remember?

Reply:
Haha. Yep, this is because `input_data["target"] = np.zeros(input_data.index.size, dtype=bool)`, the boolean for target column was initialized to be `False`. So with `DataFrame.where()` we check if the values are not <0, then we switch them as `True`.

As it is not straight-forward (especially for a quick first bite 😄 ), I refactor it as:

```py
        input_data["target"] = np.ones(input_data.index.size, dtype=bool)
        input_data["target"] = input_data["target"].where(
            input_data["i_interval"] > 0, other=False
        )
```